rooms.js

```diff
@@ -933,16 +933,34 @@ class BattleRoom extends Room {
 	}
 	requestKickInactive(user, force) {
 		if (this.resetTimer) {
-			if (user) this.sendUser(user, '|inactive|The inactivity timer is already counting down.');
-			return false;
+			let timerSetByUser = user && (this.resetUser === user.userid || (user in this.game.players && this.resetUser === '+'));
```

Comment:
I get what he's saying, but I'm wondering is whether he actually is going
to do the refactor first or not.